### ISSUE TITLE
Pandas interface for reading files

### ISSUE THREAD

**Issue by tcfuji** [igraph/python-igraph#29]
Since the R-version of igraph makes reading files (for example, in csv format) easy, could the Python version use something similar? Something like `igraph.read_csv`?

**Comment by ntamas** [comment]
You can read graphs in CSV format into igraph like this:

```
import csv, igraph
reader = csv.reader(open("input.csv"))
graph = igraph.Graph.TupleList(reader)
```

Or, if the CSV file contains headers:

```
reader = csv.DictReader(open("input.csv"))
graph = igraph.Graph.DictList(vertices=None, edges=reader)
```

Of course there is the possibility of wrapping these in a `Graph.Read_CSV` method, but there are too many possibilities with the interpretation of the columns in the CSV file (e.g., are there headers? which columns contain the endpoints of the edges? is the graph directed? which remaining columns should be stored as edge attributes? how can one provide attributes for the vertices? which vertex attribute should be used to store the names of the vertices?) that I believe it is hard to make a CSV reader that caters for all these use-cases, and it is yet unclear to me what is the "core" functionality that should be covered by the function. (The truth is, I don't really use CSV files to store my data). If you happen to have a good idea of what should be handled by `Graph.Read_CSV` and which of the above questions are not relevant for more than 20% of the use-cases, I'm happy to discuss this further.

**Comment by bill10** [comment]
any progress? The solution from @ntamas works for reading in an edge list but does not work as general as the "graph_from_data_frame" counterpart in R. If there are isolates in the network then this solution won't be able to read in the isolates as they won't show up in the edgelist. Also, the "graph_from_data_frame" in R can read in node attributes and edge attributes easily.

**Comment by szhorvat** [comment]
Does this help? https://igraph.discourse.group/t/how-to-create-a-graph-from-pandas-dataframes/34

**Comment by iosonofabio** [comment]
We have now implemented ways of reading from pandas dataframes, so you can read the CSV into pandas and then come into igraph that way.

Since CSV is not a standardized graph format, reinventing the wheel and also designing new file formats seems like a questionable endeavour given the pandas support.

So I'm going to close this one.

**Comment by ntamas** [comment]
@iosonofabio The CI build is failing with a recent version of Pandas, see [here](https://travis-ci.org/github/igraph/python-igraph/jobs/723406293). As I am not familiar with Pandas, can you please take a look at what's wrong there?

**Comment by iosonofabio** [comment]
Yep, I found it too. It was a combination of a few small typos. I included the fixes in #310 - which now passes CI happily. So I'd say when that goes through we can close this one again too.
